### PR TITLE
Trivial cleanups Jan 2021

### PR DESCRIPTION
**Pull request by Lolligerhans**
- Using `if constexpr` for Tracing (and in `E::pieces`). Does not change binary since pretty obvious for the compiler, but might as well have it. I left `if (Pt == KNIGHT || Pt == BISHOP)` at normal `if`. Otherwise it triggers define-but-not-used warning. Personally, I like having it but I can revert this change to rely on the compiler.
- Creating header _psqt.h_ for _psqt.cpp_. _main.cpp_ and _position.h_ no longer have their own declarations for `::PSQT`. This changes binary on my build. I assume it is harmless though. I copied the leading copyright note `1:1` from another file.
- Updating the _psqt.cpp_ Bonus comment which was no longer up to date and misleading.
- Since I am touching _psqt.cpp_ anyway; removing macro `S`, indenting with 2 for consistency within the same file.

Open for suggestions.

Merge/pull with `--squash`.

---

You can PR onto branch `trivial-cleanups_Jan2021` in my fork of Stockfish (https://github.com/Lolligerhans/Stockfish/tree/trivial-cleanups_Jan2021) to have me add your trivial cleanups. For simple changes, comment below and I add them for you.

:eyes: = I have seen your comment
:+1: = Comment is resolved or discussed further below

### CODE REVIEW

**Comment by ddobbelaere** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/movegen.cpp#L260 missing space after 'underpromotions'

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/evaluate.cpp#L45

missing t after 'Microsof'

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/evaluate.cpp#L63
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/evaluate.cpp#L65

An acronym such as NNUE is typically spelled with capital letters.

**Comment by cweissmz** [comment]
Should the comment in 

https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/material.cpp#L48

be 'THEIR PIECES' since lines 43-52 refer to QuadraticTheirs while the following comment

https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/material.cpp#L37

is from QuadraticOurs in lines 32-41?

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/movegen.cpp#L208

Missing i in 'instantiations'

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/position.cpp#L556

Missing - in pseudo-legal

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/movepick.h#L114
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/movepick.h#L115

Missing - in alpha-beta

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/position.cpp#L1070

The acronym SEE should be spelled with capital letters.

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/tune.h#L133

Missing i in facility (i suppose...).

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/syzygy/tbprobe.cpp#L1003

appears to link to an article called 'On the Implementation of Minimum Redundancy Prefix Codes' however the link seems to be non-functional. The following link appears to link to one of the author's profiles at researchgate and to be functional

https://www.researchgate.net/profile/Andrew_Turpin/publication/3159499_On_the_implementation_of_minimum_redundancy_prefix_codes/links/0deec51d0e74f8c73e000000.pdf

**Comment by cweissmz** [comment]
https://github.com/official-stockfish/Stockfish/blob/ee3f7b6b6e1a1051b32cedb38ac89b3458ff4ab2/src/syzygy/tbprobe.cpp#L1443

Missing - in '50-move-counter'

**Comment by MichaelB7** [comment]
Is there a way we can make these recommendations without generating an excessive amount of emails?

excessive

**Comment by BM123499** [comment]
@MichaelB7 check your notification setting on github, there you can choose if you want to receive those e-mails.

**Comment by MichaelB7** [comment]
It was a rhetorical question, I should I have not received 10 emails for 1 commit/patch, I should get one email.  It appears  that something went amissed in the PR.  Not looking for further discussion, just wanted to call that out to the submitter of #3301 @cweissmz so perhaps we avoid it in the future.  I am sure it was not intentional.

**Comment by MichaelB7** [comment]
_Also, on  on /// **pseudo legal**. It is used to validate moves from TT that can be corrupted_ , no hyphen is required when the double word adjectives follows the noun , this was discussed in the last patch that had a similar issue #3294

**Comment by Lolligerhans** [comment]
@cweissmz I believe the `QuadraticTheirs` array is for one of our pieces + one of theirs. The case of two of their pieces is handled by `QuadraticOurs` when called for their side. I added an explaining comment.

**Comment by BM123499** [comment]
@Lolligerhans Can you fix the comment above `QUIET_CHECK` movegen. it doesn't mention that only generate checking quiet. And it doesn't mention that checking castle is excluded.

File: src/movegen.cpp [hunk 1]
```diff
@@ -280,8 +280,8 @@ template ExtMove* generate<QUIETS>(const Position&, ExtMove*);
 template ExtMove* generate<NON_EVASIONS>(const Position&, ExtMove*);
 
 
-/// generate<QUIET_CHECKS> generates all pseudo-legal non-captures.
-/// Returns a pointer to the end of the move list.
+/// generate<QUIET_CHECKS> generates all pseudo-legal non-captures giving check,
+/// except castling. Returns a pointer to the end of the move list.
```
[the remark this answers is not in the document]
Comment (BM123499) [reply src/movegen.cpp:284]: it's okay for me.

**Comment by vondele** [comment]
@Lolligerhans I'll merge this later today. Much appreciated if your create the next PR to collect such changes.

**Comment by manyoso** [comment]
To be clear, this change did cause a functional change. The bench numbers changed.

**Comment by vondele** [comment]
I don't think so, there were 4 patches committed to master, one was a functional change. Not this one as far as I can see.

**Comment by manyoso** [comment]
yes, i was wrong. got confused. sorry guys.

**Pull request closed without merging**